engine/dlib/src/dlib/hash.cpp

```diff
@@ -666,24 +666,31 @@ DM_DLLEXPORT void dmHashReverseErase64(uint64_t hash)
 DM_DLLEXPORT const char* dmHashReverseSafe64(uint64_t hash)
 {
     const char* s = (const char*)dmHashReverse64(hash, 0);
-    if (s == 0)
-    {
-        char tmp[64];
-        dmSnPrintf(tmp, sizeof(tmp), "<unknown:%llu>", (unsigned long long)hash);
-        return tmp;
-    }
-    return s;
+    return s != 0 ? s : "<unknown>";
 }
 
 DM_DLLEXPORT const char* dmHashReverseSafe32(uint32_t hash)
 {
     const char* s = (const char*)dmHashReverse32(hash, 0);
-    if (s == 0)
-    {
-        char tmp[32];
-        dmSnPrintf(tmp, sizeof(tmp), "<unknown:%u>", hash);
-        return tmp;
-    }
-    return s;
+    return s != 0 ? s : "<unknown>";
 }
 
+DM_DLLEXPORT const char* dmHashReverseSafeBuffer64(uint64_t hash, char* buffer, uint32_t buffer_size)
+{
+    const char* s = (const char*)dmHashReverse64(hash, 0);
+    if (s)
+        dmStrlCpy(buffer, s, buffer_size);
+    else
+        dmSnPrintf(buffer, buffer_size, "<unknown:%llu>", hash);
```

Comment:
hmm
tests fails on `:%llu`

maybe we should use `PRIu64 ` from `<inttypes.h>` ?
like
```
dmSnPrintf(buffer, buffer_size, "<unknown:%" PRIu64 ">", hash);
```
?

Reply:
We have the DM_HASH_FMT for these cases, and unless you object, I can use that instead.

Currently, it's hex format, but I can add an integer version.
Although, it'd be good if ther are consistent throughout.

Reply:
yes, sure. I think it doesn't matter what exactly to convert into a hash in a NE  (trying to get string in Debug version)